python/cudf/cudf/core/column/string.py

```diff
@@ -2053,34 +2053,40 @@ def _nbytes(self):
 
     def as_numerical_column(self, dtype, **kwargs):
 
-        mem_dtype = np.dtype(dtype)
-        str_dtype = mem_dtype
-        out_dtype = mem_dtype
+        out_dtype = np.dtype(dtype)
+        kwargs.update(dtype=out_dtype)
 
-        if mem_dtype.type is np.datetime64:
+        if out_dtype.type is np.datetime64:
             if "format" not in kwargs:
                 if len(self) > 0:
                     # infer on host from the first not na element
                     fmt = datetime.infer_format(self[self.notna()][0])
                     kwargs.update(format=fmt)
-        kwargs.update(dtype=out_dtype)
 
-        if str_dtype.kind in ("i"):
+            # Check for None strings
+            if len(self) > 0 and self.binary_operator("eq", "None").any():
+                raise ValueError("Could not convert `None` value to datetime")
 
+            boolean_match = self.binary_operator("eq", "NaT")
+        elif out_dtype.kind in ("i"):
             if not cpp_is_integer(self).all():
                 raise ValueError(
                     "Could not convert strings to integer \
                         type due to presence of non-integer values."
                 )
-        elif str_dtype.kind in ("f"):
-
+        elif out_dtype.kind in ("f"):
             if not cpp_is_float(self).all():
                 raise ValueError(
                     "Could not convert strings to float \
                         type due to presence of non-floating values."
                 )
 
-        return _str_to_numeric_typecast_functions[str_dtype](self, **kwargs)
+        result_col = _str_to_numeric_typecast_functions[out_dtype](
+            self, **kwargs
+        )
+        if (out_dtype.type is np.datetime64) and boolean_match.any():
+            result_col[boolean_match] = None
```

Comment:
`NaT` values are the datetime equivalent of `NaN` values where we should be able to control if they get converted to `null` or stay as `NaT` if we support them. That being said, `NaT` is a numpy / Pandas specific thing.

For example, Arrow doesn't have a concept of `NaT`. I would be in strong support of not supporting `NaT` other than in going from string --> datetime64 and then converting it to null always.

Any thoughts?

Reply:
> `That being said, `NaT` is a numpy / Pandas specific thing.

+1

> 
> For example, Arrow doesn't have a concept of `NaT`. I would be in strong support of not supporting `NaT` other than in going from string --> datetime64 and then converting it to null always.
> 
> Any thoughts?

Agree on this, I think we should stick to Arrow behavior except handling "NaT" in string to datetime64 conversions, but I don't see any other flows in which "NaT" can be introduced?
For example we already handle `NaT` as `null` when we convert a pandas object to cudf.

```python
>>> cudf.from_pandas(pd.Series(["2001-01-01", "2002-02-02", "2000-01-05", "NaT"]).astype('datetime64[s]'))
0   2001-01-01
1   2002-02-02
2   2000-01-05
3         null
dtype: datetime64[ns]
```